homeassistant/components/compit/__init__.py

```diff
@@ -0,0 +1,60 @@
+"""The Compit integration."""
+
+from __future__ import annotations
+
+import asyncio
+import logging
+
+from compit_inext_api import CompitAPI, DeviceDefinitionsLoader
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import DOMAIN, PLATFORMS
+from .coordinator import CompitDataUpdateCoordinator
+
+_LOGGER: logging.Logger = logging.getLogger(__package__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Compit from a config entry."""
+
+    session = async_get_clientsession(hass)
+    api = CompitAPI(entry.data["email"], entry.data["password"], session)
+    gates = await api.authenticate()
+    device_definitions = await DeviceDefinitionsLoader.get_device_definitions(
+        hass.config.language
+    )
+
+    if gates is False:
+        return False
```

Comment:
```suggestion
    if gates is False:
        return False

    device_definitions = await DeviceDefinitionsLoader.get_device_definitions(
        hass.config.language
    )

```
since `device_definitions` is not used if `gates` is False

Reply:
You are right!